### PR TITLE
gometalinter and .travis.yml

### PR DESCRIPTION
**Pull request by jokeyrhyme**
- hooked up `make lint` to gometalinter, tweaked settings, fixed warnings
- added `make test` even though there aren't any tests yet
- created a .travis.yml file to run `make lint` and `make test` when Travis CI is enabled for this repo

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -70,7 +70,11 @@ type App struct {
 }
 
 func (a *App) Run(ctx context.Context) (retErr error) {
-	defer a.Logger.Sync()
+	defer func() {
+		if err := a.Logger.Sync(); err != nil {
+			a.Logger.Error("Unable to flush logs", zap.Error(err))
+		}
+	}()
```
Comment (jokeyrhyme) [inline app/app.go]: There was an unhandled error here, although I'm not sure I've handled this effectively

File: http.go [hunk 2]
```diff
@@ -25,7 +27,9 @@ func StartStopTLSServer(ctx context.Context, srv *http.Server, shutdownTimeout t
 		c, cancel := context.WithTimeout(context.Background(), shutdownTimeout)
 		defer cancel()
 		if srv.Shutdown(c) != nil {
-			srv.Close()
+			if err := srv.Close(); err != nil {
+				logz.LoggerStr("default", "default").Error("Server listener failed to close", zap.Error(err))
+			}
```
Comment (jokeyrhyme) [inline http.go]: Another unhandled error, and another potentially clumsy attempt at handling it

File: work_queue.go [hunk 3]
```diff
@@ -52,10 +52,6 @@ func (q *workQueue) addRateLimited(item gvkQueueKey) {
 	q.queue.AddRateLimited(item)
 }
 
-func (q *workQueue) addAfter(item gvkQueueKey, duration time.Duration) {
-	q.queue.AddAfter(item, duration)
-}
-
```
Comment (jokeyrhyme) [inline work_queue.go:58]: Apparently this was unused code

File: app/app.go (hunk 1 above)
Comment (ash2k) [reply app/app.go]: If it cannot flush logs then it probably cannot log :) This is unnecessary.

File: http.go (hunk 2 above)
Comment (ash2k) [reply http.go]: This is also not needed - the program is shutting down and if for some reason the http server couldn't cleanly shut down it is not a big deal. Remove please.

**Review by ash2k — DISMISSED** [review]

File: Makefile [hunk 4]
```diff
@@ -3,8 +3,18 @@ ALL_GO_FILES=$$(find . -type f -name '*.go' -not -path "./vendor/*")
 .PHONY: setup
 setup:
 	go get -u golang.org/x/tools/cmd/goimports
+	go get -u github.com/alecthomas/gometalinter
```
Comment (scottgreenup) [inline Makefile]: Should we be using something like gopkg.in to get the latest stable version?
https://gopkg.in/alecthomas/gometalinter.v2

Comment (jokeyrhyme) [reply Makefile]: @scottgreenup : I think @amckague tried this recently and it complicated things

Comment (amckague) [reply Makefile]: It's worth a try. I didn't dig into what broke just that it did.

**Comment by jokeyrhyme** [comment]
@scottgreenup @amckague switched to gopkg.in/alecthomas/gometalinter.v2 , seems to work just fine here

**Review by ash2k — APPROVED** [review]

**Comment by ash2k** [comment]
Squash-merge please

**Pull request merged**